src/NServiceBus.Core/Recoverability/RecoverabilityContext.cs

```diff
@@ -16,14 +16,25 @@ class RecoverabilityContext : BehaviorContext, IRecoverabilityContext, IRecovera
             RecoverabilityAction recoverabilityAction,
             IBehaviorContext parent) : base(parent)
         {
-            ErrorContext = errorContext;
+            FailedMessage = errorContext.Message;
+            Exception = errorContext.Exception;
+            ReceiveAddress = errorContext.ReceiveAddress;
+            ImmediateProcessingFailures = errorContext.ImmediateProcessingFailures;
+
             RecoverabilityConfiguration = recoverabilityConfig;
             Metadata = metadata;
             RecoverabilityAction = recoverabilityAction;
+
             Extensions.Set(errorContext.TransportTransaction);
```

Comment:
`ErrorContext` also has an `Extensions` bag in addition to the `TransportTransaction` do we need to merge the Extensions bag as well?

Reply:
It is done here https://github.com/Particular/NServiceBus/blob/master/src/NServiceBus.Core/Recoverability/RecoverabilityPipelineExecutor.cs#L37